pkg/scalers/azure/azure_eventhub_checkpoint.go

```diff
@@ -224,8 +224,22 @@ func (checkpointer *defaultCheckpointer) extractCheckpoint(get *azblob.DownloadR
 }
 
 func getCheckpoint(ctx context.Context, httpClient util.HTTPDoer, info EventHubInfo, checkpointer checkpointer) (Checkpoint, error) {
+	var podIdentity = info.PodIdentity
+
+	// For back-compat, prefer a connection string over pod identity when present
+	if len(info.StorageConnection) != 0 {
+		podIdentity.Provider = kedav1alpha1.PodIdentityProviderNone
+	}
```

Comment:
This is a misconfiguration, IMO we should log at least a warning because otherwise, users could think they are using manage identities, and they are using the connection string.

Reply:
Let me know if you think it's worth adding a logger to this module to add a log event.

Reply:
@JorTurFer , could you comment on the remarks above?

As to e2e test: not sure if that can be achieved with the current approach given that it requires a real pod-identity to be a meaningful test. Could you suggest an approach if you have one in mind?

Reply:
I'm going to resolve this, due to no follow up. @JorTurFer, please reopen if you feel there's anything left to address.